### ISSUE TITLE
Use Momentum-based gradient descent for Sunglasses Detection Neural Networks

### ISSUE THREAD

**Issue by ayushi2626** [UppuluriKalyani/ML-Nexus#36]
Use momentum-based gradient descent to help optimize the process. 

Momentum-based gradient descent helps to accelerate toward the direction of steepest descent. It helps to counter issues with vanilla gradient descent like oscillations, getting stuck at local minima and slow convergence.

**Comment by UppuluriKalyani** [comment]
@ayushi2626 Go for it!

**Comment by ayushi2626** [comment]
Hey I had a small doubt, so the code starts by downloading a bunch of pictures from the website then trains on that. After we save the model, it uses another drive folder for sample pictures right, so these sample pictures are different from the original data it has trained?

**Comment by ayushi2626** [comment]
I had tweaked certain parameters and now for the training the accuracy is 97 while the accuracy on the sample dataset is 52

**Comment by UppuluriKalyani** [comment]
> Hey I had a small doubt, so the code starts by downloading a bunch of pictures from the website then trains on that. After we save the model, it uses another drive folder for sample pictures right, so these sample pictures are different from the original data it has trained?

Yes both training and testing data are different

**Comment by UppuluriKalyani** [comment]
> I had tweaked certain parameters and now for the training the accuracy is 97 while the accuracy on the sample dataset is 52

Does it predicting correctly not for all images atleast 75% correct?

**Comment by ayushi2626** [comment]
For the training data that is taken from the website: https://www.cs.cmu.edu/afs/cs.cmu.edu/project/theo-8/faceimages/faces/
the accuracy is 97.3%. After saving the model and using it on the sample dataset from the drive, the accuracy is 52%. 

For which dataset do we want to increase the accuracy to 75%?

**Comment by UppuluriKalyani** [comment]
@ayushi2626 Please push the changes once so that I can say where should I want good accuracy
